cpp/src/replace/replace.cu

```diff
@@ -861,6 +863,49 @@ std::unique_ptr<cudf::column> replace_nulls(cudf::column_view const& input,
 }  // namespace experimental
 }  // namespace cudf
 
+namespace cudf {
+namespace experimental {
+namespace detail {
+
+std::unique_ptr<column> replace_nans(column_view const& input,
+                                     column_view const& replacement,
+                                     cudaStream_t stream,
+                                     rmm::mr::device_memory_resource* mr)
+{
+  auto nan_mask = is_not_nan(input, rmm::mr::get_default_resource(), stream);
```

Comment:
Instead of materializing the boolean column, how about using the `copy_if_else` detail variant that takes an iterator and use a transform iterator that checks `is_nan`. Shouldn't be much more code, while being a good deal more efficient.

Reply:
It's nice approach, +1